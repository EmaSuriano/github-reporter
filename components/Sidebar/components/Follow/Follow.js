import React from 'react';
import { element, number, string } from 'prop-types';
import { Box, Text } from 'grommet';

const Follow = ({ icon, title, quantity }) => (
  <Box align="center" gap="medium" direction="row">
    <Text>{icon}</Text>
    <Box align="center">
      <Text>{quantity}</Text>
      <Text>{title}</Text>
    </Box>
  </Box>
);

Follow.defaultProps = {
  quantity: 0,
};

Follow.propTypes = {
  icon: element.isRequired,
  title: string.isRequired,
  quantity: number,
};

export default Follow;
