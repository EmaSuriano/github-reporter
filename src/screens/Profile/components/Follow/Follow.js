import React from "react";
import { element, number, string } from "prop-types";

const Follow = ({ icon, title, quantity }) => (
  <div className="follow-box">
    <span className="follow-icon">{icon}</span>
    <h3 className="follow-number">{quantity}</h3>
    <span className="follow-text">{title}</span>
  </div>
);

Follow.defaultProps = {
  quantity: 0
};

Follow.propTypes = {
  icon: element.isRequired,
  quantity: number,
  title: string.isRequired
};

export default Follow;
