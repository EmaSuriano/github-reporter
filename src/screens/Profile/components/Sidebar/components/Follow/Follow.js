import React from "react";
import { element, number, string } from "prop-types";

const Follow = ({ icon, title, quantity }) => (
  <div className="follow-box">
    <span className="follow-icon">{icon}</span>
    <div className="follow-stat">
      <h3 className="follow-stat--number">{quantity}</h3>
      <span className="follow-stat--text">{title}</span>
    </div>
  </div>
);

Follow.defaultProps = {
  quantity: 0
};

Follow.propTypes = {
  icon: element.isRequired,
  quantity: number,
  title: string.isRequired
};

export default Follow;
