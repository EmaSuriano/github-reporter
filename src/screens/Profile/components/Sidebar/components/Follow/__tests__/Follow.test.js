import React from "react";
import { shallow } from "enzyme";

import Follow from "../Follow";

const props = {
  icon: <span>Icon</span>,
  title: "Title",
  quantity: 10
};

describe("Follow Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Follow {...props} />);
  });

  describe("render", () => {
    it("should be match with html structure of the snapshot", () =>
      expect(wrapper).toMatchSnapshot());
  });
});
