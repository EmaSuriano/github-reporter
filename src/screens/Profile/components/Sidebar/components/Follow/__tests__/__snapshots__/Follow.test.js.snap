// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Follow Component render should be match with html structure of the snapshot 1`] = `
<div
  className="follow-box"
>
  <span
    className="follow-icon"
  >
    <span>
      Icon
    </span>
  </span>
  <div
    className="follow-stat"
  >
    <h3
      className="follow-stat--number"
    >
      10
    </h3>
    <span
      className="follow-stat--text"
    >
      Title
    </span>
  </div>
</div>
`;
